subsys/bluetooth/host/gatt.c

```diff
@@ -2641,6 +2644,17 @@ uint8_t bt_gatt_check_perm(struct bt_conn *conn, const struct bt_gatt_attr *attr
 	}
 
 	mask &= attr->perm;
+
+	if (mask & BT_GATT_PERM_128B_ENTROPY_MASK) {
+#if defined(CONFIG_BT_SMP)
+		if ((conn->le.keys->flags & BT_KEYS_SC) == 0) {
+			return BT_ATT_ERR_INSUFFICIENT_ENCRYPTION;
+		}
+#else
```

Comment:
What about the OOB here?

Reply:
@Thalley , I think that's a good idea. Do you have any hint on how we can actually check this?

Reply:
> @Thalley , I think that's a good idea. Do you have any hint on how we can actually check this?

I don't think we have any way of doing that right now. 
One approach would be to follow the approach of `BT_KEYS_SC` by adding a `BT_KEYS_OOB` and then do a similar check. 

But before we go ahead with that, we should proably verify that an OOB key always provide 128-bit entropy (like LESC), or if it just means that it is possible.

Reply:
Maybe someone with more knowledge can chip in on this topic, e.g. @Rasmus-Abildgren-Bose?
But I believe that the OOB does not strictly provide a 128b entropy key. OOB is only an association method, and can also be used for Legacy Pairing.

Regardless, if we we are missing the functionality to actually check OOB key today, wouldn't such increased functionality fit better in a future PR? Especially if what I wrote above is true. The implications of that could be that we need to provide an API function specifically for 128b entropy keys or better.

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/46923#discussion_r911791990

Reply:
@fredrikdanebjer What the plan to deal with OOB here? Another commit in this PR, or open an issue in Github? 

Since this is a change to a public API, and the change is technically incorrect relative to the API documentation, this should definitely be fixed ASAP (preferably in this PR IMO).

Reply:
The plan is another open issue in Github. I believe this becomes quite a big change to add at this point.
Is it preferable that I create such an issue, referring to this PR before it is closed?

Reply:
> The plan is another open issue in Github. I believe this becomes quite a big change to add at this point. Is it preferable that I create such an issue, referring to this PR before it is closed?

It is preferable that it is fixed as part of this PR, as otherwise the documentation is basically lying ;) If you do create a new issue and fix it in a new PR, I would ask that you start working on it immediately. 

@jhedberg What do you think?

Reply:
I'm ok doing it in parts, but could we add a short note to the documentation mentioning that OOB is not yet covered?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/47600 should satisfy the OOB requirement